src/Nethermind/Nethermind.Synchronization.Test/SyncProgressResolverTests.cs

```diff
@@ -178,85 +143,50 @@ public void Is_fast_block_bodies_finished_returns_false_when_blocks_not_download
-            IReceiptStorage receiptStorage = Substitute.For<IReceiptStorage>();
-            IDb stateDb = new MemDb();
-            SyncConfig syncConfig = new()
-            {
-                FastBlocks = true,
-                DownloadBodiesInFastSync = false,
-                DownloadReceiptsInFastSync = true,
-                PivotNumber = "1",
-            };
-            ProgressTracker progressTracker = new(blockTree, stateDb, LimboLogs.Instance);
-
-            blockTree.LowestInsertedHeader.Returns(Build.A.BlockHeader.WithNumber(1).WithStateRoot(TestItem.KeccakA).TestObject);
-            blockTree.LowestInsertedBodyNumber.Returns(2);
-            receiptStorage.LowestInsertedReceiptBlockNumber.Returns(1);
 
-            SyncProgressResolver syncProgressResolver = new(blockTree, receiptStorage, stateDb, NullTrieNodeResolver.Instance, progressTracker, syncConfig, LimboLogs.Instance);
-            Assert.True(syncProgressResolver.IsFastBlocksBodiesFinished());
-        }
-
-        [Test]
-        public void Is_fast_block_receipts_finished_returns_true_when_receipts_not_downloaded_and_we_do_not_want_to_download_receipts()
+        private SyncProgressResolver CreateProgressResolver(IBlockTree blockTree, IDb stateDb, NullTrieNodeResolver nodeResolver, bool isReceiptFinished, SyncConfig syncConfig, LimboLogs limboLogs)
         {
-            IBlockTree blockTree = Substitute.For<IBlockTree>();
-            IReceiptStorage receiptStorage = Substitute.For<IReceiptStorage>();
-            IDb stateDb = new MemDb();
-            SyncConfig syncConfig = new()
-            {
-                FastBlocks = true,
-                DownloadBodiesInFastSync = true,
-                DownloadReceiptsInFastSync = false,
-                PivotNumber = "1",
-            };
-            ProgressTracker progressTracker = new(blockTree, stateDb, LimboLogs.Instance);
-
-            blockTree.LowestInsertedHeader.Returns(Build.A.BlockHeader.WithNumber(1).WithStateRoot(TestItem.KeccakA).TestObject);
-            blockTree.LowestInsertedBodyNumber.Returns(1);
-            receiptStorage.LowestInsertedReceiptBlockNumber.Returns(2);
-
-            SyncProgressResolver syncProgressResolver = new(
-                blockTree, receiptStorage, stateDb, NullTrieNodeResolver.Instance, progressTracker, syncConfig, LimboLogs.Instance);
-            Assert.True(syncProgressResolver.IsFastBlocksReceiptsFinished());
+            ISyncFeed<ReceiptsSyncBatch?> receiptFeed = Substitute.For<ISyncFeed<ReceiptsSyncBatch?>>();
+            receiptFeed.IsFinished.Returns(isReceiptFinished);
+
+            return new SyncProgressResolver(
+                blockTree,
+                new FullStateFinder(blockTree, stateDb, nodeResolver),
+                syncConfig,
+                Substitute.For<ISyncFeed<HeadersSyncBatch?>>(),
+                Substitute.For<ISyncFeed<BodiesSyncBatch?>>(),
+                Substitute.For<ISyncFeed<ReceiptsSyncBatch?>>(),
```

Comment:
you define the `receiptFeed` above and set the return value for `IsFinished` and then you dont use it down

Reply:
Fixed.